solutions/USACO jan19-icy-perimeter.mdx

```diff
@@ -0,0 +1,93 @@
+---
+id: usaco-895
+source: USACO Silver 2019 January
+title: Icy Perimeter
+author: Tanish Tyagi 
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_perimeter_silver_jan19.html)
+
+<LanguageSection>
+
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+//created by Tanish Tyagi
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+
+const int MAXN = 1005;
+
+char grid[MAXN][MAXN]; // the grid itself
+int n;
+bool visited[MAXN][MAXN]; // keeps track of which nodes have been visited
+ll currA = 0;
+ll currP = 0;
+ll area = INT_MIN;
+ll perimeter = INT_MIN;
+
+void setIO(string name, bool includeout = false) { // name is nonempty for USACO file I/O
+	ios_base::sync_with_stdio(0); cin.tie(0); // see Fast Input & Output
+	// alternatively, cin.tie(0)->sync_with_stdio(0);
+	freopen((name + ".in").c_str(), "r", stdin); // see Input & Output
+	if (includeout) {
+		freopen((name + ".out").c_str(), "w", stdout);
+	}
+}
+
+void floodfill(int r, int c) {
+	if (r < 0 || r >= n || c < 0 || c >= n) {
+		currP++;
+		return; // if outside grid
+	}
+	if (visited[r][c]) return; // already visited this square
+	if (grid[r][c] == '.') {
+		currP++;
+		return;
+	}
+	visited[r][c] = true; // mark current square as visited
+	if (grid[r][c] == '#') {
+		currA++;
+	}
+	// recursively call floodfill for neighboring squares
+	floodfill(r, c + 1);
+	floodfill(r, c - 1);
+	floodfill(r - 1, c);
+	floodfill(r + 1, c);
+}
+
+int main() {
+	setIO("perimeter", true);
```

Comment:
change to 
`freopen("perimeter.in", "r", stdin);`
and
`freopen("perimeter.out", "w", stdout);`

Reply:
the setIO() function is kinda weird so you should make it concise like this :stuck_out_tongue: